### DISCUSSION TITLE
This site can’t be reached - I'm stuck

### DISCUSSION THREAD

**Discussion by ghost** [leokhoa/laragon#437] [Q&A]
Hi guys,

I'm new to Github and was kindly directed here by Pen-y-Fan (thanks again, Mike).

I've used Laragon for a year now without any issues but now I can't access my site. I'm not a programmer and hope some of you more experienced coders can help me out.

I run Laragon 5 with:
* Apache httpd-2.4.47-win64-VS16
* mysql-5.7.33-winx64
* php-7.4.19-Win32-vc15-x64

The problem occurred when I upgraded from php-7.4.19-Win32-vc15-x64 to php-8.2.3-Win32-vs16-x64. After that I got the message "This site can’t be reached".

I then removed php-8.2.3-Win32-vs16-x64 and "rolled back" to php-7.4.19-Win32-vc15-x64 again. But now still get the same message.

Nothing else changed.

I read all kinds of stuff about what might be wrong, but it's simply beyond my level :) I'm not experienced at all in php, apache and mysql. And now feel pretty stuck.

Do you have any ideas how to solve this problem?

All help will be highly appreciated.

Best,
Pete/DKturtle

**Comment by Pen-y-Fan** [comment]
Laragon 5 has some compatibility problems with PHP 8, it can work but needs some additional setup. The project may need some upgrading to work with PHP 8 too.

Normally to change PHP version in Laragon the PHP version can be switched. See my blog [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/#switch-versions-in-laragon) which has a section on how to switch versions.

Once PHP 7.4 is selected click Start all. Laragon should show Apache and MySQL started. Click the **Web** button to view the Laragon website.

Once Laragon is working as expected, try your project's website.

Laragon menu > www > project name

If Laragon or the project still doesn't display try log off and back on. If that doesn't work try a restart. This will clear any error states.

Let us know how you get on.

**Comment by ghost** [comment]
Laragon website works (Localhost):

Apache/2.4.47 (Win64) OpenSSL/1.1.1k PHP/7.4.19
PHP version: 7.4.19 [info](http://localhost/?q=info)
Document Root: C:/xxxxxx/laragon/www
[Getting Started](https://laragon.org/docs)

But after Laragon menu > www > project name I still get "This site can't be reached".

![Laragon-app](https://user-images.githubusercontent.com/127690594/224759059-dc2954a2-13ce-498c-9c70-7baa46e65319.png)
![Laragon-localhost](https://user-images.githubusercontent.com/127690594/224759930-8d22f816-27d0-4388-9dd8-bbae6456f522.gif)

**Reply by Pen-y-Fan** [reply]
Ok, Laragon is working 🎉

Before doing anything further take a backup of the project.

When PHP was upgraded, was the project changed? E.g. Was `composer update` run?

Was anything else changed in Laragon? E.g. is the preferences hostname still {name}.test?

![preferences](https://user-images.githubusercontent.com/40126936/224788545-7b0eda7c-ffb9-46c3-abd7-9b30b1ad5824.png)

- do you still have the original installation files?  
    - if so compare the original files with the current ones.
- do you have the original installation guide?
    - if so re-check the project against the original settings.
- was the project backed up before attempting the upgrade?
    - if so restore the backup.

**Reply by Pen-y-Fan** [reply]
One thing to try in Laragon, re-register the SSL by Adding Laragon.crt to the trusted store.

I had an odd experience with one site which worked on **http**://project.test, but not **https**://project.test

- See [How to set up Laragon on a new Windows computer (part 4) - Enable security #Add the certificate to the trusted store](https://pen-y-fan.github.io/2022/11/10/how-to-set-up-laragon-on-a-new-windows-computer-part-4/#add-the-certificate-to-the-trusted-store)

**Comment by ghost** [comment]
@Pen-y-Fan I fiddled a bit with what you suggested and then decided to restore the project via a backup I luckily had made before the problems occurred. That seems to have fixed the issue! Now I just wanna update Laragon to ver. 6 with php 8 (php-8.0.28-Win32-vs16-x64).

Hope this will work without further unforseen challenges :)

Thanks a lot for your fast support and help, man!

**Reply by Pen-y-Fan** [reply]
I would suggest before you upgrade Laragon v6 and PHP 8.1 download and test PHP 7.4 (7.4.33) from [windows PHP download](https://windows.php.net/download/). See [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/). The main thing to check is for deprecation warnings, these will need to be fixed before changing to PHP 8.

**Comment by ghost** [comment]
Alright, thanks. I'll follow those guidelines and make sure everything works properly and are up to date/supported (not deprecated).

**Comment by thechoperpilot** [comment]
Two months later, here I am with the same problem. Out of nowhere, I started getting the same "This site can’t be reached" error, but only for newly created websites (domains). In my case, problem was that "Auto-create virtual hosts" was not checked (not sure how this happened, I know I didn't change anything there recently). 
Turning it on again, everything started working just fine.

Thanks for your help, @Pen-y-Fan !!

**Comment by clementmas** [comment]
I also couldn't connect anymore until I realized my browser VPN was on. Don't forget to check for that.

**Comment by Stepup3** [comment]
Hello guys,

I'm new to Github and had to sign up in a bit to solve a problem.

I installed Laragon on my laptop and created a test website but now I can't access my site. I am just a freshman and hope some of you more experienced coders can help me out. This is the error message I usually get:
Fatal error: Uncaught mysqli_sql_exception: Table 'digiskillshub.wp_options' doesn't exist in C:\laragon\www\digiskillshub\wp-includes\wp-db.php:2056 Stack trace: #0 C:\laragon\www\digiskillshub\wp-includes\wp-db.php(2056): mysqli_query(Object(mysqli), 'SELECT option_n...') #1 C:\laragon\www\digiskillshub\wp-includes\wp-db.php(1945): wpdb->_do_query('SELECT option_n...') #2 C:\laragon\www\digiskillshub\wp-includes\wp-db.php(2695): wpdb->query('SELECT option_n...') #3 C:\laragon\www\digiskillshub\wp-includes\option.php(284): wpdb->get_results('SELECT option_n...') #4 C:\laragon\www\digiskillshub\wp-includes\functions.php(1730): wp_load_alloptions() #5 C:\laragon\www\digiskillshub\wp-includes\load.php(752): is_blog_installed() #6 C:\laragon\www\digiskillshub\wp-settings.php(159): wp_not_installed() #7 C:\laragon\www\digiskillshub\wp-config.php(103): require_once('C:\\laragon\\www\\...') #8 C:\laragon\www\digiskillshub\wp-load.php(50): require_once('C:\\laragon\\www\\...') #9 C:\laragon\www\digiskillshub\wp-blog-header.php(13): require_once('C:\\laragon\\www\\...') #10 C:\laragon\www\digiskillshub\index.php(17): require('C:\\laragon\\www\\...') #11 {main} thrown in C:\laragon\www\digiskillshub\wp-includes\wp-db.php on line 2056

Do you have any ideas how to solve this problem?

All help will be highly appreciated.
Stepup3
Thanks

**Comment by phu-huy** [comment]
I am working with laragon 6 installation version, using with windows 11 computer version 23H2
When I installed wordpress with default option an error occurred!
1. I tried starting windows, creating a project and reloading the server many times but it didn't work.
2. I am trying a php version 7.4.19. Still not working
3. I tried running nginx server according to the default version. Still not working
At any time, there is a solution that will help me and those who come after me. Laragon has not had any updates or patches in the last 2 years! Looks like this is a bug
![image](https://github.com/leokhoa/laragon/assets/95569554/9b33b406-49b5-4923-8e55-fe80083b30c6)
![image](https://github.com/leokhoa/laragon/assets/95569554/899f0878-3973-499d-b113-959058772b12)
![image](https://github.com/leokhoa/laragon/assets/95569554/2b3d7e6b-77e0-41db-b36b-5ca7063fc08d)

**Comment by ghost** [comment]
Which version of Laragon are you using? The current Laragon Full 6 already comes with php 8.1
Have you configured ngix too? By default, Laragon for Windows comes with Apache - that's why nginx, etc. is listed under Advanced under Services

**Comment by stefanroehrbein** [comment]
Had similar issues while configuring my new notebook with a fresh Laragon and for me it was to add the following to the http.conf:
```
<IfModule mpm_winnt_module>
    ThreadStackSize 8388608
</IfModule>
```
I was working with this for years on my old notebook and totally forgot about this - now it took my a whole day to realize i did not do this on the new one :)

**Reply by leokhoa** [reply]
@stefanroehrbein That's interesting to know! Your app must be quite large to require an 8MB Thread Stack Size. I'll add this as a default in the next version of Laragon, so no one need to worry about setting it manually.

**Reply by stefanroehrbein** [reply]
@leokhoa actually it's "just" a TYPO3 v10 (quite old) :)
Maybe it would be also sufficient to use less than 8MB... TYPO3 has a built-in environment testing tool and on windows it (always) shows this:
```
Windows apache thread stack size
This current value cannot be checked by the system, so please ignore this warning if it is already taken care of:
Fluid uses complex regular expressions which require a lot of stack space during the first processing. 
On Windows the default stack size for Apache is a lot smaller than on UNIX.
You can increase the size to 8MB (default on UNIX) by adding the following configuration to httpd.conf and restarting Apache afterwards:
<IfModule mpm_winnt_module>ThreadStackSize 8388608</IfModule>
```
That's why i just copied this over... and it does the job!

**Reply by AWqxKAWERbXo** [reply]
I'll close the discussion because the original author is gone.
